Lexical_analyzer.py

```diff
@@ -62,25 +65,48 @@ def analyze(text):
 
     tokens = []
 
+    end = re.compile(r'.*\n')
+    lines = [0]
+    for m in re.finditer(end, original_text):
+        lines.append(m.end())
+
     i = 0
     while i < len(text):
 
         current_matches = []
         for regex, token_type in rules:
             m = regex.match(text, i)
             if m:
-                current_matches.append((m, token_type))
+                current_matches.append((m, token_type, m.span()))
```

Comment:
i think it would be better to add this data in the Token class, so it will be available throughout the entire program (lexical, syntax, semantics)
and so to calculate it only for the selected token